controller/src/beerocks/master/tasks/dynamic_channel_selection_task.h

```diff
@@ -0,0 +1,118 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * Copyright (c) 2016-2019 Intel Corporation
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#ifndef _DYNAMIC_CHANNEL_SELECTION_TASK_H_
+#define _DYNAMIC_CHANNEL_SELECTION_TASK_H_
+
+#include "../db/db.h"
+#include "task.h"
+#include "task_pool.h"
+
+#include <beerocks/tlvf/beerocks_message.h>
+
+#include <chrono>
+
+// Helper MACROs for Enum/String generation
+#define GENERATE_ENUM(ENUM) ENUM,
+#define GENERATE_STRING(STRING) #STRING,
+
+namespace son {
+
+class dynamic_channel_selection_task : public task, public std::enable_shared_from_this<task> {
+public:
+    dynamic_channel_selection_task(db &database, ieee1905_1::CmduMessageTx &cmdu_tx_,
+                                   task_pool &tasks_, std::string radio_mac_);
+    virtual ~dynamic_channel_selection_task() {}
+
+    typedef struct {
+        sMacAddr radio_mac;
```

Comment:
No need for `typedef` - you can declare the struct as is and access it without the `struct` keyword in c++

Reply:
done